platform/net/firewall_provider_linux.go

```diff
@@ -45,8 +54,13 @@ func SetupNatsFirewall(mbus string) error {
 	if mbus == "" || strings.HasPrefix(mbus, "https://") {
 		return nil
 	}
+	// NOBLE_TODO: check if warden does not hit this cgroup v2 code path
+	var cgroupV2 bool
+	if cgroups.Mode() == cgroups.Unified {
+		cgroupV2 = true
+	}
 	_, err := cgroups.V1()
-	if err != nil {
+	if err != nil && !cgroupV2 {
```

Comment:
It might make this method less confusing to move the 
```golang
	_, err := cgroups.V1()
	if err != nil {
	//snip
	}
```

section down into the `SetupIptables()` func. This would allow the check at the end of this func to be:

```golang
	if cgroups.Mode() == cgroups.Unified {
		return SetupNFTables(host, port)
	} else {
		return SetupIptables(host, port, addr_array)
	}
```

... and for `cgroups.V1()` to only be checked inside `SetupIptables()` where it is required.

Reply:
change already made